### DISCUSSION TITLE
How to Hide empty regions

### DISCUSSION THREAD

**Discussion by Schuchmann** [WerWolv/ImHex#1235] [Get Help]
Hi,
I am loading an Intel Hex File which starts at Address 0x8020000.
ImHex displays the Adresses vom 0x0 to 0x8020000 with question marks:
![image](https://github.com/WerWolv/ImHex/assets/75126737/b37ee426-c805-4b21-aa9f-1c1e4ba487ff)

Is it somehow possible to hide the empty regions in ImHex?

**Comment by paxcut** [comment]
I don't think there is a way to hide some portion of the file as if it is not there without changing the file contents. You can use remove (right click on hex display and select remove) some number of bytes from any starting address, but be aware that this operation cannot be undone. 

You can use insert or fill (right click on hex display and select Insert or fill) to restore the deleted part but I recommend saving a backup just in case before doing any alterations.

**Reply by Schuchmann** [reply]
Hi paxcut,
 "don't think there is a way to hide some portion of the file as if it is not there..."
 Actually it is not there, It contains no data in the Intel Hex File, therefore I cannot
 delete it.
 Viewing Hex Files like this is unusable since you have to scroll. It would be better to have
 only the sections which are filled with data in the Intel Hex File...

**Reply by WerWolv** [reply]
Hey, there's currently no way to do this but I might look into adding it. What you can do for now is open the interesting part in a new View by selecting it and choosing the right option in the right click menu

**Reply by Schuchmann** [reply]
@WerWolv This would be great. I think this would be a good enhancement. e.g. a View like this would be nice:
![image](https://github.com/WerWolv/ImHex/assets/75126737/f0d9bd1a-9883-4076-903b-90d2b5b88c10)

Maybe it would be easy to set the used Blocks from the Intel Hex automatically as bookmarks.

**Comment by paxcut** [comment]
Also instead of scrolling use File>Goto to jump to any address.
